draft-ietf-jsonpath-base.md

```diff
@@ -911,8 +943,129 @@ size doesn't
 fit in the implementation's representation of an integer, the implementation
 MUST raise an error.
 
+### Descendant Selector
+
+#### Syntax
+{: numbered="false" toc="exclude"}
+
+The descendant selector starts with a double dot `'..'`. It can be used in place of the normal `dot-selector`. So it will usually be followed by an object member name. As an extension to that natural rule it might also be followed by an `index-selector` acting on objects or arrays.
+
+~~~~ abnf
+descendant-selector = ".." ( dot-member-name      /  ; ..<name>
+                             index-selector       /  ; ..[<index>]
+                             index-wild-selector  /  ; ..[*]
+                             "*"                     ; ..*
+                           )
+~~~~
+
+#### Semantics
+{: numbered="false" toc="exclude"}
+
+The `descendant-selector` is inspired by ECMAScript for XML (E4X). It selects object members and array elements in a recursive descent process.
+
+### Union Selector
+
+#### Syntax
+
+The union selector is syntactically related to the `index-selector`. It contains multiple, comma separated entries.
+
+~~~~ abnf
+union-selector = "[" ws union-entry 1*(ws "," ws union-enty) ws "]"
+
+union-entry    =  ( quoted-member-name / 
+                    element-index      /
+                    slice-index
+                  )
+~~~~
+
+#### Semantics
+
+A union selects any node which is selected by at least one of the union selectors and selects the concatenation of the
+lists (in the order of the selectors) of nodes selected by the union elements.
+Possibly duplicate selected nodes are kept in the node list.
```

Comment:
```suggestion
Any duplicated selected nodes are kept in the node list.
```

Reply:
Define word duplicate (or preferable don't use it but describe what happens).

Reply:
Resolved in #102